src/com/boxboat/jenkins/library/buildVersions/GitBuildVersions.groovy

```diff
@@ -28,40 +28,92 @@ class GitBuildVersions implements Serializable {
+            """
+        }
+        Map<String, Object> data = new LinkedHashMap<>()
+        data.put("version", version)
+        if(metadata != null && !metadata.isEmpty()) {
+            data.put("metadata", metadata)
+        }
+        def yamlStr = YamlUtils.dump(data)
         Config.pipeline.sh """
             mkdir -p "${dir}"
-            echo "${version}" > "${path}"
         """
+        Config.pipeline.writeFile(file: yamlFile, text: yamlStr, encoding: "Utf8")
     }
 
     String getEventImageVersion(String event, Image image) {
         def dir = "${gitRepo.dir}/image-versions/${event}"
-        def path = "${dir}/${Utils.alphaNumericDashLower(image.path)}.txt"
-        def result = Config.pipeline.sh(returnStdout: true, script: """
-            if [ -f "${path}" ]; then
-                cat "${path}"
-            fi
-        """)?.trim()
+        def path = "${dir}/${Utils.alphaNumericDashLower(image.path)}"
+        def txtFile = "${path}.txt"
+        def yamlFile = "${path}.yaml"
+        def result = ""
+
+        if (Config.pipeline.fileExists(txtFile)) {
+            result = Config.pipeline.sh(returnStdout: true, script: """
+                cat "${txtFile}"
+            """)?.trim()
+        } else if (Config.pipeline.fileExists(yamlFile)) {
+            String yamlVersion = Config.pipeline.sh(returnStdout: true, script: """
+                cat "${yamlFile}"
+            """)?.trim()
+            Map<Object, Object> yamlData = YamlUtils.load(yamlVersion) as Map<Object, Object>
+            result = yamlData.getOrDefault("version", "")
+        }
+
         return Utils.resultOrTest(result, testVersion(event))
     }
 
+    Map<Object, Object> getEventImageVersionMetadata(String event, Image image) {
+        def dir = "${gitRepo.dir}/image-versions/${event}"
+        def path = "${dir}/${Utils.alphaNumericDashLower(image.path)}"
+        def yamlFile = "${path}.yaml"
+
+        if (Config.pipeline.fileExists(yamlFile)) {
+            String yamlVersion = Config.pipeline.sh(returnStdout: true, script: """
+                cat "${yamlFile}"
+            """)?.trim()
+            Map<Object, Object> yamlData = YamlUtils.load(yamlVersion) as Map<Object, Object>
+            return yamlData.getOrDefault("metadata", new LinkedHashMap<Object, Object>()) as Map<Object, Object>
+        }
+        return null
+    }
+
+    Map<Object, Object> addEventImageVersionMetadata(String event, Image image, Map<Object, Object> imageMetadataMap) {
+        Map<Object, Object> metaMap = new LinkedHashMap<>()
+        metaMap.putAll(imageMetadataMap)
+        def meta = getEventImageVersionMetadata(event, image)
+
+        def tag_key = "image_tag_${Utils.alphaNumericUnderscoreLower(image.path)}"
+        if (meta != null){
+            Map<Object, Object> imageMetaData = new LinkedHashMap<>()
+            imageMetadataMap.put("metadata", meta)
+            metaMap.put(tag_key, imageMetadataMap)
+        }
+        return metaMap
```

Comment:
I'm having a hard time following what's going on here, and I don't see direct usages of this method. Can you explain?

Reply:
Actually this method isn't being used - I meant to delete it